keyserver/listener.go

```diff
@@ -10,11 +10,16 @@ import (
 	. "github.com/coniks-sys/coniks-go/protocol"
 )
 
-func (server *ConiksServer) listenForRequests(ln *net.TCPListener, handler func(msg []byte) ([]byte, error)) {
+func (server *ConiksServer) listenForRequests(ln net.Listener, handler func(msg []byte) ([]byte, error)) {
```

Comment:
Hmm, the type switches do seem a bit clunky, but I'm not sure if separating the listeners would really give us a significant performance improvement. Maybe that way there would be some parallelism would could take advantage of?